lib/load-plugin.js

```diff
@@ -31,12 +32,32 @@ function loadPlugin (instance, plugin, callback) {
   // place the plugin at the top of _current
   instance._current.unshift(plugin)
 
-  plugin.exec((last && last.server) || instance._server, (err) => {
+  if (instance._error && !plugin.isAfter) {
```

Comment:
I moved this from plugin.exec, as it seemed odd to me, that we could do this already in the loadPlugin function, and decouple so Plugin from Avvio.